### ISSUE TITLE
False Positive S1656/no-self-assignment: spread operator on array

### ISSUE THREAD

**Issue by andrea-guarino-sonarsource** [SonarSource/SonarTS#760]
```typescript
let array = [1,2,3]
// ....
array = [...array] // FP
```

Reassigning value of `array` by performing a copy (via spread operator) has the effect to change its reference.

This is sometimes used by change detection frameworks in a MVC application to trigger a _View_ refresh. (See examples listed [here](https://github.com/SonarSource/SonarTS/issues/753))
